sdk/testing/src/main/java/io/opentelemetry/sdk/testing/assertj/SpanDataAssert.java

```diff
@@ -455,6 +455,13 @@ public SpanDataAssert hasStatus(StatusData status) {
     return this;
   }
 
+  /** Asserts the span has a status satisfying the given condition. */
+  public SpanDataAssert hasStatusSatisfying(Consumer<StatusData> condition) {
```

Comment:
Thanks @HaloFour - can you instead add `StatusAssert` with things like `hasCode`, `hasDescription`, and other helpers possibly `hasDescriptionMatching(String regex)`? And this can be `Consumer<StatusAssert>`. We've settled on using that pattern and find it is simpler and more fluent then just consumers on the data objects.